src/pip/_internal/cli/index_command.py

```diff
@@ -26,7 +27,8 @@
 
 def _create_truststore_ssl_context() -> Optional["SSLContext"]:
     if sys.version_info < (3, 10):
-        raise CommandError("The truststore feature is only available for Python 3.10+")
+        logger.warning("Disabling truststore because Python version isn't 3.10+")
+        return None
```

Comment:
![image](https://github.com/pypa/pip/assets/63936253/107c1fb1-c363-4493-9ad1-79cf20c990b2)

To be honest, emitting a warning on every single invocation of pip on Python 3.9 and lower is plain annoying. This can be a debug or verbose log.

Reply:
Yea, this should definitely not be a warning or info log. Verbose or debug sounds reasonable to me!

Reply:
FWIW, CI is failing on <3.10 because of this superflous warning as well. :)

Reply:
Thanks for fixing this @pradyunsg! :purple_heart: